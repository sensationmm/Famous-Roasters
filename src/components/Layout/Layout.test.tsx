import { render, waitFor } from '@testing-library/react'
import React from 'react'

import { Layout } from '.'

describe('Layout component', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <Layout>
        <span>Test</span>
      </Layout>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
