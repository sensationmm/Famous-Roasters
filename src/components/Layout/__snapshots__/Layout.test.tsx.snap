// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Renders correctly 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col"
  >
    <div
      class="bg-white"
    >
      <header
        class="relative bg-white"
      >
        <nav
          aria-label="Top"
          class="max-w-7xl mx-auto px-4 sm:px-6 xl:px-8"
        >
          <div
            class="h-16 flex items-center justify-between border-b border-brand-grey-whisper"
          >
            <div
              class="flex-1 flex items-center xl:hidden"
            >
              <button
                class="-ml-2 bg-white p-2 rounded-md text-gray-400"
                data-testid="button-open"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Open menu
                </span>
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 6h16M4 12h16M4 18h16"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="hidden xl:flex-1 xl:block xl:self-stretch"
            >
              <div
                class="h-full flex space-x-8"
              >
                <a
                  class="flex items-center text-gray-700 hover:text-gray-800"
                  href="/taste-finder"
                >
                  <span
                    class="uppercase text-sm leading-5 font-normal"
                  >
                    Tastefinder
                  </span>
                </a>
                <a
                  class="flex items-center text-gray-700 hover:text-gray-800"
                  href="/our-roasters"
                >
                  <span
                    class="uppercase text-sm leading-5 font-normal"
                  >
                    Unsere Röster
                  </span>
                </a>
              </div>
            </div>
            <a
              class="flex h-full items-center pl-2 pr-2"
              href="/"
            >
              <span
                class="sr-only"
              >
                Famous Roasters
              </span>
              <div
                class="xl:hidden font-syne text-3xl"
              >
                <span>
                  F
                </span>
                <span
                  class="font-extrabold"
                >
                  R
                </span>
              </div>
              <div
                class="hidden xl:block font-syne text-xl text-center"
              >
                <div
                  class="h-5"
                >
                  Famous
                </div>
                <div
                  class="font-extrabold"
                >
                  Roasters
                </div>
              </div>
            </a>
            <div
              class="flex-1 flow-root flex h-full items-center"
            >
              <div
                class="flex-1 flex h-full items-center justify-end xl:flex-none"
              >
                <div
                  class="hidden h-full xl:flex space-x-8 justify-end"
                >
                  <a
                    class="flex items-center text-gray-700 hover:text-gray-800"
                    href="/about"
                  >
                    <span
                      class="uppercase text-sm leading-5 font-normal"
                    >
                      Über uns
                    </span>
                  </a>
                  <a
                    class="flex items-center text-gray-700 hover:text-gray-800"
                    href="/contact"
                  >
                    <span
                      class="uppercase text-sm leading-5 font-normal"
                    >
                      Kontakt
                    </span>
                  </a>
                </div>
                <a
                  class="hidden xl:block p-2 text-gray-400 hover:text-gray-500 xl:ml-6"
                  href="/account"
                >
                  <span
                    class="sr-only"
                  >
                    Konto
                  </span>
                  <svg
                    aria-hidden="true"
                    class="w-6 h-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </a>
                <div
                  class="ml-4 flow-root xl:ml-6"
                >
                  <a
                    class="group -m-2 p-2 flex items-center"
                    href="/cart"
                  >
                    <svg
                      aria-hidden="true"
                      class="flex-shrink-0 h-6 w-6 text-gray-400 group-hover:text-gray-500"
                      fill="none"
                      stroke="currentColor"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 11V7a4 4 0 00-8 0v4M5 9h14l1 12H4L5 9z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                    <span
                      class="ml-2 text-sm font-medium text-gray-700 group-hover:text-gray-800"
                    >
                      0
                    </span>
                    <span
                      class="sr-only"
                    >
                      items in cart, view bag
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </nav>
      </header>
    </div>
    <span>
      Test
    </span>
    <footer
      class="bg-brand-black text-base text-white px-6 sm:px-6 xl:px-10 py-10"
    >
      <div
        class="w-full text-center mb-10 flex justify-center items-center"
      >
        <div>
          [Badge]
        </div>
        <div
          class="text-5xl leading-none font-normal"
        >
          A pleasure for every taste.
        </div>
      </div>
      <div
        class="max-w-7xl mx-auto grid gap-2 gap-y-10 grid-cols-1 md:grid-cols-2"
      >
        <div
          class="order-2 md:order-1"
        >
          <ul>
            <li
              class="mt-0"
            >
              <a
                href="/catalogue"
              >
                Shop
              </a>
            </li>
            <li
              class="mt-6"
            >
              <a
                href="/taste-finder"
              >
                Tastefinder
              </a>
            </li>
            <li
              class="mt-6"
            >
              <a
                href="/our-roasters"
              >
                Unsere Röster
              </a>
            </li>
            <li
              class="mt-6"
            >
              <a
                href="/about"
              >
                Über uns
              </a>
            </li>
            <li
              class="mt-6"
            >
              <a
                href="/jobs"
              >
                Jobs
              </a>
            </li>
            <li
              class="mt-6"
            >
              <a
                href="/privacy"
              >
                Datenschutz-Bestimmungen
              </a>
            </li>
            <li
              class="mt-6"
            >
              <a
                href="/press"
              >
                Presse
              </a>
            </li>
            <li
              class="mt-6"
            >
              <a
                href="/imprint"
              >
                Impressum
              </a>
            </li>
          </ul>
        </div>
        <div
          class="order-1 md:order-2"
        >
          <div>
            <div>
              <span
                class="font-syne text-3xl leading-8 font-semibold tracking-tight"
              >
                Kunden Support
              </span>
            </div>
            <div
              class="mt-6"
            >
              <span
                class="text-lg leading-7 font-normal"
              >
                Fragen? Kommentare? Bedenken? Oder möchtest du uns einfach ein paar schöne Worte senden?
              </span>
            </div>
            <div
              class="mt-6"
            >
              <button
                class="inline-flex items-center px-4 py-2 text-base text-black bg-brand-green-club rounded-full"
                role="button"
                type="button"
              >
                Kunden Support
              </button>
            </div>
          </div>
          <div
            class="mt-12"
          >
            <div>
              <span
                class="font-syne text-3xl leading-8 font-semibold tracking-tight"
              >
                Newsletter
              </span>
            </div>
            <div
              class="mt-6"
            >
              <span
                class="text-lg leading-7 font-normal"
              >
                Melde dich für den Newsletter an und erhalte, einmalige Angebote, Neuigkeiten, Tipps und Insights zum Thema Specialty Coffee und neuen Röstern. 
              </span>
            </div>
            <div
              class="mt-6"
            >
              <input
                type="text"
              />
            </div>
            <div
              class="mt-6"
            >
              <button
                class="inline-flex items-center px-4 py-2 text-base text-black bg-brand-green-club rounded-full"
                role="button"
                type="button"
              >
                Anmelden
              </button>
            </div>
            <div
              class="mt-4"
            >
              <span
                class="text-brand-grey-bombay text-base leading-6 font-normal"
              >
                Hinweis: Sie können sich jederzeit abmelden. Siehe unsere Datenschutzrichtlinie und Bedingungen.
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full text-center mt-10"
      >
        <span
          class="text-center text-base leading-6 font-normal"
        >
          © 2022 Famous Roasters. Alle Rechte vorbehalten
        </span>
        <br />
        <span
          class="text-center text-base leading-6 font-normal"
        >
          Geröstet mit ❤️ in Berlin
        </span>
      </div>
    </footer>
  </div>
</div>
`;
